### PR TITLE
spec: opensuse support

### PR DESCRIPTION
**Pull request by rjfd**
This PR adds some conditionals to the spec file to support the packaging in openSUSE based distros.
It also cleans up a bit the spec file, and also deletes python interpreter shebangs in non-executable python files.

Signed-off-by: Ricardo Dias <rdias@suse.com>

### CODE REVIEW

**Comment by rjfd** [comment]
@ktdreyer could you check if I didn't break rhel/centos and fedora packaging with these changes?

File: ceph-iscsi.spec [hunk 1]
```diff
@@ -81,31 +90,69 @@ Grafana.
...
 install -m 0644 gwcli.8 %{buildroot}%{_mandir}/man8/
 gzip %{buildroot}%{_mandir}/man8/gwcli.8
+%if 0%{?suse_version}
+mkdir -p %{buildroot}%{_unitdir}/rbd-target-gw.service.d
+install -m 0644 .%{_sysconfdir}/systemd/system/rbd-target-gw.service.d/dependencies.conf %{buildroot}%{_unitdir}/rbd-target-gw.service.d/
+mkdir -p %{buildroot}%{_sbindir}
+ln -s service %{buildroot}%{_sbindir}/rcrbd-target-gw
+ln -s service %{buildroot}%{_sbindir}/rcrbd-target-api
+%else
+mkdir -p %{buildroot}%{_sysconfdir}/systemd/system/rbd-target-gw.service.d
```
Comment (ktdreyer) [inline ceph-iscsi.spec]: Oh, yes this is a general bug in our systemd packaging here. We should install to `%{_unitdir}` on all platforms and not write files into `/etc/systemd`. Are you ok making this change for everything?

We'll still need the `suse_version` conditional here (and below in `%files`) for the rc* scripts.

File: ceph_iscsi_config/backstore.py [hunk 2]
```diff
@@ -1,4 +1,2 @@
-#!/usr/bin/env python
-
```
Comment (smithfarm) [inline ceph_iscsi_config/backstore.py]: drop this line so the file doesn't start with a blank line?

File: ceph-iscsi.spec (hunk 1 above)
Comment (rjfd) [reply ceph-iscsi.spec]: Sure, I'll fix this.

**Comment by rjfd** [comment]
@ktdreyer @smithfarm addressed your comments.

**Pull request merged**